### ISSUE TITLE
`allowAdditions` in Search

### ISSUE THREAD

**Issue by coxjc** [Semantic-Org/Semantic-UI-React#4146]
## Feature Request

I'd like to add the `allowAdditions` functionality, which is in `Dropdown`, to `Search`.

### Problem description
If there isn't an option in a dropdown, we currently have the option to have an addition. I got led from `Dropdown` to `Search` due to a ridiculously large dataset size. I think it'd be very useful to have `allowAdditions` in search as well. 

### Proposed solution
Simply put, I want to copy the functionality of `allowAdditions` in Dropdown to Search.

**Before I begin implementing this, are there any objections from the owners of the project?**

**Comment by coxjc** [comment]
[hello](https://media4.giphy.com/media/l3q2PqfeGzcZuzQTS/200.gif?cid=239532917jp17qv0u1fu90ir3oj76jximf8lzrt5annmnust&rid=200.gif)
